extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoTestBase.java

```diff
@@ -42,77 +36,47 @@ protected static String getProperty(String name) {
 
     @BeforeAll
     public static void startMongoDatabase() throws IOException {
-        try {
-            //JDK bug workaround
-            //https://github.com/quarkusio/quarkus/issues/14424
-            //force class init to prevent possible deadlock when done by mongo threads
-            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
-        } catch (ClassNotFoundException e) {
-        }
+        fixIssue14424();
+
         String uri = getConfiguredConnectionString();
         // This switch allow testing against a running mongo database.
         if (uri == null) {
             Version.Main version = Version.Main.V4_0;
             int port = 27018;
             LOGGER.infof("Starting Mongo %s on port %s", version, port);
-            MongodConfig config = MongodConfig.builder()
-                    .version(version)
-                    .net(new Net("127.0.0.1", port, false))
-                    .cmdOptions(MongoCmdOptions.builder()
-                            .useNoJournal(false)
-                            .build())
-                    .build();
-            MONGO = getMongodExecutable(config);
-            try {
-                MONGO.start();
-            } catch (Exception e) {
-                //every so often mongo fails to start on CI runs
-                //see if this helps
-                try {
-                    Thread.sleep(1000);
-                } catch (InterruptedException ignore) {
 
-                }
-                MONGO.start();
-            }
+            TransitionWalker.ReachedState<RunningMongodProcess> started = Mongod.instance()
+              .withNet(Start.to(Net.class).initializedWith(Net.builder()
+                .from(Net.defaults())
+                .port(port)
+                .build()))
+              .withMongodArguments(Start.to(MongodArguments.class)
+                .initializedWith(MongodArguments.defaults().withUseNoJournal(false)))
+              .start(version);
```

Comment:
We previously starts twice to cover startup issues that often occurs in CI, maybe it'll be good to keep this

Reply:
@loicmathieu i did a lot of refactoring with this new version.. one side effect of this refactoring: if something fails, it should tell us exactly what and why .. and in any case it cleans up, so nothing temporary is left.. if it still fails in CI, than it would be very interesting why.. and then we can fix that.. the old code base was not this kind of deterministic ..

Reply:
OK, can you lanch it multiple times on your fork to be sure it always pass ? If it pases 5 successful CI builds we can merge it as is and act later if needed.
I'll trigger a build on the Quarkus repo soon.